packages/list_bloc/lib/src/cubits/list_cubit.dart

```diff
@@ -21,4 +21,50 @@ class ListCubit<T, F> extends DataCubit<List<T>, F> {
     }
   }
 // TODO(alexis): implement remove, add, replace etc
+
+  Future<void> remove(T item) async {
+    try {
+      final data = <T>[...state.data ?? []];
+      if (data.isNotEmpty) {
+        data.remove(item);
+        emit(
+          Data(data: data, filter: state.filter),
+        );
+      } else {
+        emit(Data.empty(filter: state.filter));
+      }
+    } catch (e) {
+      emit(Data.error(data: state.data, filter: state.filter, error: e));
+      rethrow;
+    }
+  }
+
+  Future<void> add(T item) async {
```

Comment:
Why is it returning a Future

Reply:
Thank you for the review
I have made all the changes you can recheck again

Reply:
Future is not needed there cause its not perform any asynchronous task so i have remove it in the latest Pr